lib/hexpm_web/controllers/web_auth_controller.ex

```diff
@@ -0,0 +1,56 @@
+defmodule HexpmWeb.WebAuthController do
+  use HexpmWeb, :controller
+  @moduledoc false
+
+  alias Hexpm.Accounts.WebAuth
+
+  # Controller for Web Auth, a means of authenticating the cli from the website
+
+  plug :requires_login when action == :show
+
+  # step one of device flow
+  def code(conn, %{"scope" => scope}) do
+    case WebAuth.get_code(scope) do
+      {:error, _changeset} ->
+        conn
+        |> put_status(:unprocessable_entity)
+        |> json(%{"error" => "invalid scope"})
+
+      response ->
+        conn
+        |> put_status(:ok)
+        |> json(response)
+    end
+  end
+
+  def show(conn, _) do
+    render(conn, "show.html")
+  end
+
+  def submit(conn, %{"user_code" => user_code}) do
+    audit = audit_data(conn)
+    user = conn.assigns.current_user
+
+    case WebAuth.submit(user, user_code, audit) do
+      {:ok, _changeset} ->
+        conn
+        |> put_status(:ok)
+        |> redirect(to: Routes.web_auth_path(conn, :success))
```

Comment:
just double-checking, is it intentional to redirect with HTTP 200 as opposed to HTTP 302?

Reply:
No it isn't.